src/public/App.jsx

```diff
@@ -68,10 +69,17 @@ class App extends Component {
 
   render () {
```

Comment:
In case you were asking, it is exactly that kind of component that is very beautiful with some HOC:
The main component has one and only responsibility, and HOC functions composition let it be a component that response to _loading_ and _error_ states, without adding anymore responsibility to the source component.